tests/pso_test.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file pso_test.cpp
+ * @author Suryoday Basak
+ * @author Chintan Soni
+ *
+ * Test file for PSO optimizer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <ensmallen.hpp>
+#include "catch.hpp"
+
+using namespace ens;
+using namespace ens::test;
+
+TEST_CASE("LBestPSORosenbrockFunctionTest", "[PSOTest]")
+{
+  RosenbrockFunction f;
+  LBestPSO s;
+
+  arma::vec coords = f.GetInitialPoint();
+  if (!s.Optimize(f, coords))
+    FAIL("LBest PSO optimization reported failure for Rosenbrock Function.");
+
+  double finalValue = f.Evaluate(coords);
+
+  REQUIRE(finalValue == Approx(0.0).margin(1e-5));
+  REQUIRE(coords[0] == Approx(1.0).epsilon(1e-7));
+  REQUIRE(coords[1] == Approx(1.0).epsilon(1e-7));
+}
+
+TEST_CASE("LBestPSOSphereFunctionTest", "[PSOTest]")
+{
+  SphereFunction f(4);
+  LBestPSO s;
+
+  arma::vec coords = f.GetInitialPoint();
+  if (!s.Optimize(f, coords))
+    FAIL("LBest PSO optimization reported failure for Sphere Function.");
+  
+  double finalValue = f.Evaluate(coords);
+  //BOOST_REQUIRE_SMALL(result, 1e-5);
+  REQUIRE(finalValue <= 1e-5);
+  for (size_t j = 0; j < 4; ++j)
+    //BOOST_REQUIRE_SMALL(coordinates[j], 1e-3);
+	REQUIRE(coords[j] <= 1e-3); //work on this
+}
+
+
+TEST_CASE("ConstrainedSquaredFunctionTest", "[PSOTest]")
+{
+  class SquaredFunction
```

Comment:
Are you planning to switch to the other functions from: https://github.com/mlpack/ensmallen/pull/117?

Reply:
Yes, that's my plan for the coming week. I'll spend a few days working on the new test functions from PR#117. The merge was timely as I'm just beginning to make PSO run properly.